src/org/jetbrains/kotlin/test/helper/git/GitCommandFacade.kt

```diff
@@ -0,0 +1,92 @@
+﻿package org.jetbrains.kotlin.test.helper.git
+
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.progress.coroutineToIndicator
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vcs.VcsNotifier
+import git4idea.commands.Git
+import git4idea.commands.GitCommand
+import git4idea.commands.GitLineHandler
+import git4idea.repo.GitRepository
+import kotlinx.coroutines.future.await
+import org.jetbrains.kotlin.test.helper.MyBundle
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+
+@Suppress("UnstableApiUsage")
+object Git {
+    private val git by lazy { Git.getInstance() }
+    private val logger by lazy { logger<Git>() }
+    private const val GREEN_BRANCH_FETCH_URL = "https://kotl.in/green-dev-kotlin-aggregate"
+
+    context(project: Project)
+    suspend fun performFetch(repository: GitRepository, originName: String, branchName: String): Boolean =
+        coroutineToIndicator {
+            git.runCommand {
+                GitLineHandler(project, repository.root, GitCommand.FETCH).apply {
+                    addParameters(originName, branchName)
+                }
+            }.success()
+        }
+
+    context(project: Project)
+    suspend fun performUpdateRef(repository: GitRepository, revision: String, fullBranchName: String): Boolean =
+        coroutineToIndicator {
+            git.runCommand {
+                GitLineHandler(project, repository.root, GitCommand.UPDATE_REF).apply {
+                    addParameters(fullBranchName, revision)
+                }
+            }.success()
+        }
+
+    context(project: Project)
+    suspend fun performRebase(repository: GitRepository, revision: String, autoStash: Boolean = false): Boolean =
+        coroutineToIndicator {
```

Comment:
The `autoStash` parameter is unclear to me.
On the one hand, its name tells that it allows performing rebase even having some uncommitted changes in the repo.
On the other hand, if this parameter is `true`, you pass the `--autosquash` flag, which is used in interactive rebase for automatic processing of `!fixup`, `!squash` and `!amend` commit. 

Could you please bring some light on it?

Reply:
Oops, seems like a typo, thanks for noticing! It should a `--autostash` for sure. I'll fix it

Reply:
In this case I'm concerned about the fact that it's set to `true` in "rebase to green master" action.
`!fixup` commits are quite common in our dev branches which are on review, and in a lot of cases people don't want to squash them before the final merging.

Reply:
Ah, now I misread what you said :facepalm: 

> Oops, seems like a typo, thanks for noticing! It should be a --autostash for sure. I'll fix it

I read it as auto**squash**, not auto**stash**.

So, let's settle: auto**stash** was intended, and it's ok.